File: flowcore/src/main/java/com/raizlabs/android/dbflow/annotation/PrimaryKey.java
package com.raizlabs.android.dbflow.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description:
 */
@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.FIELD)
public @interface PrimaryKey {

    /**
     * Specifies if the column is autoincrementing or not
     *
     * @return the columnType int
     */
    boolean autoincrement() default false;
}

File: core/src/commonMain/kotlin/com/dbflow5/annotation/PrimaryKey.kt
package com.dbflow5.annotation

/**
 * Description:
 */
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.FIELD)
annotation class PrimaryKey(
    /**
     * Specifies if the column is autoincrementing or not
     */
    val autoincrement: Boolean = false,
    /**
     * Specifies the column to be treated as a ROWID but is not an [.autoincrement]. This
     * overrides [.autoincrement] and is mutually exclusive.
     */
    val rowID: Boolean = false,
)
